Source/Workers/decodeDraco.js

```diff
@@ -0,0 +1,120 @@
+define([
+        '../Core/arraySlice',
+        '../Core/ComponentDatatype',
+        '../Core/defined',
+        '../Core/IndexDatatype',
+        '../Core/RuntimeError',
+        '../ThirdParty/draco-decoder-gltf',
+        './createTaskProcessorWorker'
+    ], function(
+        arraySlice,
+        ComponentDatatype,
+        defined,
+        IndexDatatype,
+        RuntimeError,
+        draco,
+        createTaskProcessorWorker) {
+    'use strict';
+
+    var dracoDecoder;
+
+    function decodeIndexArray(dracoGeometry) {
+        var numPoints = dracoGeometry.num_points();
+        var numFaces = dracoGeometry.num_faces();
+
+        var faceIndices = new draco.DracoInt32Array();
+        var indexArray = IndexDatatype.createTypedArray(numPoints, numFaces * 3);
+
+        for (var i = 0; i < numFaces; ++i) {
+            dracoDecoder.GetFaceFromMesh(dracoGeometry, i, faceIndices);
+
+            var offset = i * 3;
+            indexArray[offset + 0] = faceIndices.GetValue(0);
+            indexArray[offset + 1] = faceIndices.GetValue(1);
+            indexArray[offset + 2] = faceIndices.GetValue(2);
+        }
+
+        draco.destroy(faceIndices);
+
+        return indexArray;
+    }
+
+    function decodeAttributeData(dracoGeometry, compressedAttributes) {
+        var numPoints = dracoGeometry.num_points();
+        var decodedAttributeData = {};
+        var attributeData;
+        var vertexArray;
+        for (var attributeName in compressedAttributes) {
+            if (compressedAttributes.hasOwnProperty(attributeName)) {
+                var compressedAttribute = compressedAttributes[attributeName];
+                var attribute = dracoDecoder.GetAttributeByUniqueId(dracoGeometry, compressedAttribute);
+                var numComponents = attribute.num_components();
+
+                if (attribute.data_type() === 4) {
+                    attributeData = new draco.DracoInt32Array();
+                    vertexArray = new Uint16Array(numPoints * numComponents);
```

Comment:
Do we have information about the max value in the `DracoInt32Array`? Ideally `vertexArray` should be either a `Uint16Array` or `Uint32Array` depending on the max value.

Reply:
This value isn't available through the draco API. We can only get it if we start iterating through and getting the values. Is it worth stopping and remaking the array if it should be `Uint32`?

Reply:
It's _probably_ safe to just use `uint16` then. I can't think of an integer attribute that commonly has elements above 65535. Leave a comment though explaining why a `Uint16Array` is used and the potential problem.

Reply:
OK, will do. `Int` attributes were rarely used, I don't think the glTF example models use them at all.